### PR TITLE
Code Review

### PR DESCRIPTION
**Pull request by awhitford**
- Removed obsolete `new` keywords.
 - Added types to collection and function variables.
 - Added `final` and `const` keywords.
 - Replaced `null` guards with concise `?.` and `??` operators.
 - Added missing null check on `dispose` for `FadeAnimatedTextKit`.
 - In `fade.dart`, renamed the `_RotatingTextState` class to `_FadeTextState` to be consistent with the overall pattern and avoid confusion with `_RotatingTextState` in `rotate.dart`.

**Warning**:
 - Removed `onNextBeforePause` from `ColorizeAnimatedTextKit` because it was not referenced.

## Pull Request Process

1. Ensure any install or build dependencies are removed before the end of the layer when doing a build.
2. Update the README.md if needed with details of changes to the interface, this includes new environment variables, exposed ports, useful file locations and container parameters.
3. Increase the version numbers in any examples files and the README.md to the new version that this Pull Request would represent. The versioning scheme we use is [SemVer](http://semver.org/).
4. You may merge the Pull Request in once you have the sign-off of one developer, or if you do not have permission to do that, you may request the second reviewer to merge it for you.

### CODE REVIEW

File: lib/src/colorize.dart [hunk 1]
```diff
@@ -86,7 +80,7 @@ class ColorizeAnimatedTextKit extends StatefulWidget {
       : super(key: key);
 
   @override
-  _RotatingTextState createState() => new _RotatingTextState();
+  _RotatingTextState createState() => _RotatingTextState();
```
Comment (aagarwal1012) [inline lib/src/colorize.dart:83]: Can you also rename this line?

File: lib/src/scale.dart [hunk 2]
```diff
@@ -101,7 +101,7 @@ class ScaleAnimatedTextKit extends StatefulWidget {
       : super(key: key);
 
   @override
-  _RotatingTextState createState() => new _RotatingTextState();
+  _RotatingTextState createState() => _RotatingTextState();
```
Comment (aagarwal1012) [inline lib/src/scale.dart:104]: Also, rename `_RotatingTextState ` here.

File: lib/src/fade.dart [hunk 3]
```diff
@@ -173,9 +159,8 @@ class _RotatingTextState extends State<FadeAnimatedTextKit>
                 animation: _controller,
                 builder: (BuildContext context, Widget child) {
                   return Opacity(
-                    opacity: !(_fadeIn.value == 1.0)
-                        ? _fadeIn.value
-                        : _fadeOut.value,
+                    opacity:
+                        _fadeIn.value != 1.0 ? _fadeIn.value : _fadeOut.value,
                     child: Text(
```
Comment (AadumKhor) [inline lib/src/fade.dart:164]: @awhitford Make this `Text` widget a child of the `AnimatedBuilder` and reference it using the child property given in the builder function.

**Comment by aagarwal1012** [comment]
Also, please resolve the merge conflicts.

File: lib/src/fade.dart (hunk 3 above)
Comment (awhitford) [reply lib/src/fade.dart]: Good idea.

**Comment by AadumKhor** [comment]
@awhitford Great work! Please do a squash merge.

**Comment by aagarwal1012** [comment]
@awhitford, I have sent you a collaborator invite to this repo. Great work, hope to see more contributions from you.

**Comment by aagarwal1012** [comment]
@allcontributors add @yiminghan for Ideas & Planning and Maintenance.

**Comment by aagarwal1012** [comment]
@allcontributors add @awhitford for Ideas & Planning and Maintenance.

**Pull request merged**